node/core/core.go

```diff
@@ -0,0 +1,38 @@
+package core
+
+import (
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/core"
+)
+
+// Config combines all configuration fields for Core subsystem.
+type Config struct {
+	EnableRemote bool
+	Remote       struct {
+		Protocol   string
+		RemoteAddr string
+	}
+	EmbeddedConfig *core.Config
+}
+
+// DefaultConfig returns default configuration for Core subsystem.
+func DefaultConfig() *Config {
+	return &Config{
+		EmbeddedConfig: core.DefaultConfig(),
+		EnableRemote:   false,
+	}
+}
+
+// Components collects all the components and services related to the Core node.
```

Comment:
```suggestion
// Components collects all the components and services related to communicating with a Core node
// process, whether it is embedded or remote.
```

Reply:
There will be other things besides communication, like application, let's keep this generalized.

Reply:
I am not in favor of changing comments all the time, specifically when things we develop change often. Also, we don't have an audience from the outside to use/read our code, so for now it's not a priority to make comments very descriptive.

> we don't know how the Celestia Node <> Celestia app relationship will look

We defined that in the recent ADR. Furthermore, I know that more components for the core will come soon, thus, in case I accept the suggestion, the comment must be changed accordingly once a new component is added or removed. In case we keep it general nothing would need to be changed.

Reply:
Documentation will change as implementation changes (as it should) - it's not a big deal to be more descriptive now. It will help us in the future (it's more for our own knowledge than anyone else's right now) but we also have other engineers on the team who will eventually need to use these components and won't have the same insight we do which is why I am in favour of something more descriptive. The current comment you have gives 0 insight into why there is a `Core` node there and what the celestia node is doing with it.

Reply:
```suggestion
// Components collects all the components and services related to managing the relationship with the Core node.
```